In [1]:
import pandas as pd
import numpy as np

In [4]:


df=pd.read_csv("/content/House Price India.csv")
df.head()

,id,Date,number of bedrooms,number of bathrooms,living area,lot area,number of floors,waterfront present,number of views,condition of the house,grade of the house,Area of the house(excluding basement),Area of the basement,Built Year,Renovation Year,Postal Code,Lattitude,Longitude,living_area_renov,lot_area_renov,Number of schools nearby,Distance from the airport,Price
0,6762810145,42491,5,2.50,3650,9050,2.0,0,4,5,10,3370,280,1921,0,122003,52.8645,-114.557,2880,5400,2,58,2380000
1,6762810635,42491,4,2.50,2920,4000,1.5,0,0,5,8,1910,1010,1909,0,122004,52.8878,-114.470,2470,4000,2,51,1400000
2,6762810998,42491,5,2.75,2910,9480,1.5,0,0,3,8,2910,0,1939,0,122004,52.8852,-114.468,2940,6600,1,53,1200000
3,6762812605,42491,4,2.50,3310,42998,2.0,0,0,3,9,3310,0,2001,0,122005,52.9532,-114.321,3350,42847,3,76,838000
4,6762812919,42491,3,2.00,2710,4500,1.5,0,0,4,8,1880,830,1929,0,122006,52.9047,-114.485,2060,4500,1,51,805000


In [3]:
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

In [8]:
df.columns

Index(['id', 'Date', 'number of bedrooms', 'number of bathrooms',
       'living area', 'lot area', 'number of floors', 'waterfront present',
       'number of views', 'condition of the house', 'grade of the house',
       'Area of the house(excluding basement)', 'Area of the basement',
       'Built Year', 'Renovation Year', 'Postal Code', 'Lattitude',
       'Longitude', 'living_area_renov', 'lot_area_renov',
       'Number of schools nearby', 'Distance from the airport', 'Price'],
      dtype='object')

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14620 entries, 0 to 14619
Data columns (total 23 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   id                                     14620 non-null  int64  
 1   Date                                   14620 non-null  int64  
 2   number of bedrooms                     14620 non-null  int64  
 3   number of bathrooms                    14620 non-null  float64
 4   living area                            14620 non-null  int64  
 5   lot area                               14620 non-null  int64  
 6   number of floors                       14620 non-null  float64
 7   waterfront present                     14620 non-null  int64  
 8   number of views                        14620 non-null  int64  
 9   condition of the house                 14620 non-null  int64  
 10  grade of the house                     14620 non-null  int64  
 11  Ar

In [4]:
df.describe()

,id,Date,number of bedrooms,number of bathrooms,living area,lot area,number of floors,waterfront present,number of views,condition of the house,...,Built Year,Renovation Year,Postal Code,Lattitude,Longitude,living_area_renov,lot_area_renov,Number of schools nearby,Distance from the airport,Price
count,1.462000e+04,14620.000000,14620.000000,14620.000000,14620.000000,1.462000e+04,14620.000000,14620.000000,14620.000000,14620.000000,...,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,1.462000e+04
mean,6.762821e+09,42604.538646,3.379343,2.129583,2098.262996,1.509328e+04,1.502360,0.007661,0.233105,3.430506,...,1970.926402,90.924008,122033.062244,52.792848,-114.404007,1996.702257,12753.500068,2.012244,64.950958,5.389322e+05
std,6.237575e+03,67.347991,0.938719,0.769934,928.275721,3.791962e+04,0.540239,0.087193,0.766259,0.664151,...,29.493625,416.216661,19.082418,0.137522,0.141326,691.093366,26058.414467,0.817284,8.936008,3.675324e+05
min,6.762810e+09,42491.000000,1.000000,0.500000,370.000000,5.200000e+02,1.000000,0.000000,0.000000,1.000000,...,1900.000000,0.000000,122003.000000,52.385900,-114.709000,460.000000,651.000000,1.000000,50.000000,7.800000e+04
25%,6.762815e+09,42546.000000,3.000000,1.750000,1440.000000,5.010750e+03,1.000000,0.000000,0.000000,3.000000,...,1951.000000,0.000000,122017.000000,52.707600,-114.519000,1490.000000,5097.750000,1.000000,57.000000,3.200000e+05
50%,6.762821e+09,42600.000000,3.000000,2.250000,1930.000000,7.620000e+03,1.500000,0.000000,0.000000,3.000000,...,1975.000000,0.000000,122032.000000,52.806400,-114.421000,1850.000000,7620.000000,2.000000,65.000000,4.500000e+05
75%,6.762826e+09,42662.000000,4.000000,2.500000,2570.000000,1.080000e+04,2.000000,0.000000,0.000000,4.000000,...,1997.000000,0.000000,122048.000000,52.908900,-114.315000,2380.000000,10125.000000,3.000000,73.000000,6.450000e+05
max,6.762832e+09,42734.000000,33.000000,8.000000,13540.000000,1.074218e+06,3.500000,1.000000,4.000000,5.000000,...,2015.000000,2015.000000,122072.000000,53.007600,-113.505000,6110.000000,560617.000000,3.000000,80.000000,7.700000e+06


**Q1. What is the average, median, and standard deviation of house prices?**

In [10]:
df.Price.describe()

,Price
count,1.462000e+04
mean,5.389322e+05
std,3.675324e+05
min,7.800000e+04
25%,3.200000e+05
50%,4.500000e+05
75%,6.450000e+05
max,7.700000e+06


In [14]:
mean=df.Price.mean()
median=df.Price.median()
std=df.Price.std()

print(f"Average = {mean}")
print(f"Median = {median}")
print(f"Standard Deviation = {std}")

Average = 538932.2183310534
Median = 450000.0
Standard Deviation = 367532.3808039669


**Q2. Which number of bedrooms is most common?**

In [19]:
df['number of bedrooms'].value_counts()



,count
number of bedrooms,
3,6612
4,4724
2,1844
5,1079
6,176
1,136
7,30
8,11
9,3


In [32]:
print(df['number of bedrooms'].value_counts()[:1].to_string())


number of bedrooms
3    6612


**Q3. Check if the price data is skewed.**

In [34]:
print(df.Price.skew())

4.269297720707116


**Q4. What is the average price for each number of bedrooms?**

In [35]:
df.groupby('number of bedrooms')['Price'].mean()

,Price
number of bedrooms,
1,3.089638e+05
2,3.985476e+05
3,4.632776e+05
4,6.361988e+05
5,7.752550e+05
6,8.375815e+05
7,1.016544e+06
8,1.208455e+06
9,7.766663e+05


**Q5. What is the relationship between living area and price? (Use : Area without Basement).**

In [37]:
print(df['Area of the house(excluding basement)'].corr(df['Price']))


0.6152204198678108


### 0.70 to 1.00: Strong positive relationship (highly connected)
### 0.30 to 0.69: Moderate positive relationship
### 0.00 to 0.29: Weak or no relationship.


## Here the relationship between living area and price = Moderate Positive

**Q6. Identify any anomalies where houses have high prices but low areas.**

In [15]:
# Define thresholds using the 75th percentile for price and 25th percentile for area
high_price_threshold = df['Price'].quantile(0.75)
low_area_threshold = df['living area'].quantile(0.25)

# Filter rows meeting BOTH anomaly conditions
anomalies = df[(df['Price'] > high_price_threshold) & (df['living area'] < low_area_threshold)]

# Sort them by price (ascending or descending) to see the most extreme cases
anomalies_sorted = anomalies.sort_values(by='Price', ascending=False)

# View the clean anomalies (using .head() or printing the full dataset as preferred)
print(f"Found {len(anomalies_sorted)} structural anomalies.")
print(anomalies_sorted[['Price', 'living area']].head())


Found 88 structural anomalies.
        Price  living area
12955  907000         1340
13163  903000         1140
81     900000         1330
12692  900000         1240
10922  900000         1190


**Q7. Compare average price based on number of floors and houses with or without waterfront.**

In [24]:
pivot_comparison = df.pivot_table(
    values='Price',
    index='number of floors',
    columns='waterfront present',
    aggfunc='mean')

print(pivot_comparison)

waterfront present             0             1
number of floors                              
1.0                 4.323724e+05  1.211081e+06
1.5                 5.476162e+05  1.232507e+06
2.0                 6.357742e+05  2.075931e+06
2.5                 1.087654e+06  3.280000e+06
3.0                 5.558095e+05  1.655000e+06
3.5                 1.205875e+06           NaN


**Q8. Identify the minimum and maximum house price. What does this indicate?**

In [18]:
min_price = df.Price.min()
max_price = df.Price.max()

print(f"minimum house price = {min_price}")
print(f"maximum house price = {max_price}")

minimum house price = 78000.0
maximum house price = 7060000.0


It indicates minimum house price is 78000 and maximum house price is 7060000.It means there is house from less expensive to luxury.

**Q9. Which location (zipcode/area) has the highest average price?**

In [42]:
high_avgprice=df.groupby('Postal Code')['Price'].mean().sort_values(ascending=False).head(1)
print(high_avgprice)

Postal Code
122071    2.348311e+06
Name: Price, dtype: float64


**Q10. Write at least 5 insights from your analysis.**

A-

- House with 3 bedrooms are the most common in this dataset that is 6612.

- Relationship between living area and price is 0.6152204198678108 that is moderate positive.

- The house prices are highly positively skewed that is             4.269297720707116 indicating a long tail towards higher prices.

- The minimum house price is 78000 and maximum house price is 7060000.It means there is house from less expensive to luxury.

- 122071 postal code has the highest average price that is 2.348311.
